pretyping/coercionops.ml

```diff
@@ -220,7 +220,8 @@ let class_of env sigma t =
       let { cl_param = n1 } = class_info cl in
       (t, n1, cl, u, args)
     with Not_found ->
-      let t = Tacred.hnf_constr env sigma t in
+      let t = Tacred.hnf_constr0 env sigma t in
+      let t = Reductionops.clos_whd_flags RedFlags.betaiota env sigma t in
```

Comment:
is it actually useful to do whd betaiota after hnf?

Reply:
The simpl implementation is so complex it cannot hurt. We have fast paths there, if it's already in whnf it will be a quick noop.